backend/btrixcloud/migrations/migration_0032_dupe_org_names.py

```diff
@@ -0,0 +1,79 @@
+"""
+Migration 0032 - Case-insensitive org name duplicates
+"""
+
+from uuid import UUID
+
+from pymongo.errors import DuplicateKeyError
+
+from btrixcloud.migrations import BaseMigration
+from btrixcloud.utils import slug_from_name
+
+
+MIGRATION_VERSION = "0032"
+
+
+class Migration(BaseMigration):
+    """Migration class."""
+
+    # pylint: disable=unused-argument
+    def __init__(self, mdb, **kwargs):
+        super().__init__(mdb, migration_version=MIGRATION_VERSION)
+
+    async def migrate_up(self):
+        """Perform migration up.
+
+        Check for case-insensitive duplicate org names and slugs.
+        If found, rename org/slug as necessary to avoid duplicates
+        regardless of case.
+        """
+        orgs_db = self.mdb["organizations"]
+
+        org_name_set = set()
+        org_slug_set = set()
+
+        cursor = orgs_db.find({})
+        async for org_dict in cursor:
+            name = org_dict.get("name", "")
+            slug = org_dict.get("slug", "")
+
+            rename_org = False
+
+            if name.lower() in org_name_set:
+                rename_org = True
+            else:
+                org_name_set.add(name.lower())
+
+            if slug.lower() in org_slug_set:
+                rename_org = True
+            else:
+                org_slug_set.add(slug.lower())
+
+            if rename_org:
+                await self.update_org_name_and_slug(orgs_db, name, org_dict.get("_id"))
+
+    async def update_org_name_and_slug(self, orgs_db, old_name: str, oid: UUID):
+        """Rename org"""
+        count = 1
+        suffix = f" {count}"
+
+        while True:
+            org_name = f"{old_name}{suffix}"
+            org_slug = slug_from_name(org_name)
+
+            try:
+                await orgs_db.find_one_and_update(
+                    {"_id": oid}, {"$set": {"slug": org_slug, "name": org_name}}
+                )
+                print(
+                    f"Renamed org {oid} to {org_name} with slug {org_slug}", flush=True
+                )
+                break
+            except DuplicateKeyError:
```

Comment:
Hm, will this get a DuplicateKeyError here if renaming due to case, since case-insensitive index hasn't been set yet...

Reply:
Good point, I added an extra check against the name and slug sets. Will keep this in just as a fail-safe too I suppose.

Reply:
Ah actually, I think I see, this already starts with 'Name 1' (should be 'Name 2'). For slug, we always lower-case it, so that should work. However, for name, I wonder if this could create other duplicates, eg. if we have:
'Name', 'name', 'name 2', then 'Name' gets renamed to 'Name 2', causing a case sensitive conflict with 'name 2', but it wasn't detected in the initial pass?

Reply:
Again, quite an edge-case, heh

Reply:
With the extra commit I put in to check against the set before the db update, if 'name' is renamed to 'name 2', the extra check will catch it and update it to 'name 3' before trying to commit it to the database.

Reply:
ok, yep, that makes sense! tested this as well